modules/s3/src/xtdb/s3/checkpoint_transfer_manager.clj

```diff
@@ -112,6 +112,16 @@
                                                                ::s3-dir (str s3-dir "/")
                                                                ::cp/checkpoint-at cp-at}))})
         cp)))
+  
+  (cleanup-checkpoint [this {:keys [tx cp-at]}]
```

Comment:
does this also need to delete the edn file first, so that no nodes download a corrupted checkpoint?

Reply:
I am a little unclear on what you mean - the current behaviour of both this and the same implementation of `cleanup-checkpoint` on the other s3 checkpoint store is that they delete everything _including_ the edn file?

Reply:
Or would you prefer that it splits out the edn file from the list and calls to delete that separately?

Reply:
@danmason yeah, would prefer that we confirm the EDN file's been deleted successfully before deleting anything else - we can't necessarily rely on its position in the list meaning it's deleted first, for example

Reply:
Can certainly do that :)